### DISCUSSION TITLE
Migrating from Nextcloud on Server 1 to Nextcloud-AIO on Server 2. How to?

### DISCUSSION THREAD

**Discussion by lutzmarkwirth** [nextcloud/all-in-one#2403] [Questions] [answered]
Hello all,

I am planning to move a Nextcloud. Here is the scenario:

I am running a V-Server from Strato (800 GB storage, 32 GB RAM) with Nextcloud 24.0.4, Docker, nginx as reverse proxy and MariaDB. I set it all up following these instructions: https://github.com/evertramos/docker-nextcloud-letsencrypt. My backups so far I do via the backup tool from Strato. The Nextcloud as such I have not backed up individually. There are about 240 GB of data on my Nextcloud.

Since I have not yet succeeded in integrating features such as Talk and Collabora and I also find the administration of the Nextcloud to be time-consuming, I would like to move to a Nextcloud AIO. To do this, as I have learned, I need to move to a KVM server architecture.

My planned approach is to first set everything up on the new server and briefly redirect the domain accordingly for this (due to the domain check).

What option do I have to transfer the data afterwards? Could I for example update the old Nextcloud instance and transfer the data via BorgBackup?

I am grateful and look forward to any hints.

Many thanks
Lutz

**Comment by szaimen** [comment, marked as answer]
Hi, see https://github.com/nextcloud/all-in-one/blob/main/migration.md

**Reply by Gekko23** [reply]
This is by far the worst description of an 'how-to' i can think of. It's absolutely unclear to me what the author tries to explain regarding the database migration. What variables are variables, what variables are placeholders for users to fill in their credentials? Nothing is explained, just thrown to mention, without even trying to think what non IT experts may distill from this garbage. 

It's a shame that nextcloud doesn't provide any tools to migrate natively. But it's a REAL shame, what people like this terrible author throw up and call an 'explanation'. If anyone asks about migrating a nextcloud instance to an nextcloud AIO instance, this garbage is handed over all the time, although this text is not sufficiently explaining. It doesn't even regard the wide spread use of docker containers. What a shame. What a mess. What a low effort.
